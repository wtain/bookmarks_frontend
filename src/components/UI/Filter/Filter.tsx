
import ITagsRepository from "../../../domain/repository/tags/ITagsRepository";
import TagCloud from "../TagCloud/TagCloud";
import { useNavigate } from "react-router-dom";

interface Props {
  tagsRepository: ITagsRepository;
}

const Filter = (props: Props) => {
  const navigate = useNavigate();

  return (
    <TagCloud tagsRepository={props.tagsRepository}
      onTagClick={
          (tag: string) => {
              navigate("/bookmarks/" + tag)
          }}
      onClearTagsClick={
          () => {
              navigate("/bookmarks")
          }
      }/>
    // todo: Filter closed
  )
}

export default Filter;