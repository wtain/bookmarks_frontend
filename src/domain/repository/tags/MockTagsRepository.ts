import ITagsRepository from './ITagsRepository';

class MockTagsRepository implements ITagsRepository {
  async getTags(): Promise<string[]> {
    return ["tag1", "tag2", "tag3"];
  }
  
}

export default MockTagsRepository;