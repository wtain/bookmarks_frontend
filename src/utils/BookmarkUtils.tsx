
import React from "react";
import BookmarkDto from "../domain/dto/BookmarkDto";
import {v4 as uuidv4} from 'uuid';
import TagDto from "../domain/dto/TagDto";

const createBookmark = (summary: string, contents: string, tags: TagDto[]): BookmarkDto => {
    return {
                summary, 
                contents, 
                id: uuidv4(), 
                created: new Date().toLocaleString(),
                tags,
                isDone: false
            }
}

const createNewTag = (): TagDto => {
    return { name: "", id: uuidv4() }
}

const BookmarkUtils = { createBookmark, createNewTag };

export default BookmarkUtils;