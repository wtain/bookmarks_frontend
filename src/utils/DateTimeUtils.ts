import moment from "moment";

export function parseDateTimeFromMongo(datetime: Date | undefined) {
  if (datetime === undefined) {
    return undefined;
  }
  return moment(datetime.toISOString()).fromNow();
}

export function addOneMonth(date: Date): Date {
  let result = new Date(date);
  result.setMonth(result.getMonth() + 1)
  return result;
}

export function firstDayOfMonth(date: Date): Date {
  let result = new Date(date);
  result.setDate(1)
  return result;
}

export function getDatePortion(date: Date): string {
  const dateCopy = new Date(date);
  dateCopy.setUTCHours(0);
  dateCopy.setMinutes(0);
  const momentForDate = moment(dateCopy);
  return momentForDate.format('YYYY-MM-DD');
}

export function datesEqual(date1: Date, date2: Date): boolean {
  return getDatePortion(date1) === getDatePortion(date2);
}