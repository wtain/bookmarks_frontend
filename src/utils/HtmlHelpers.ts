import sanitizeHtml from "sanitize-html";

export function sanitizeText(text: string){
  return sanitizeHtml(text, {
    allowedTags: ['br', 'b', 'a', 'img']
  });
}