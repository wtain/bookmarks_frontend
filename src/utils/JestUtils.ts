
export function jestHacks() {
    // https://stackoverflow.com/questions/39830580/jest-test-fails-typeerror-window-matchmedia-is-not-a-function
    // https://github.com/facebook/create-react-app/issues/10126
    window.matchMedia = (query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })
}