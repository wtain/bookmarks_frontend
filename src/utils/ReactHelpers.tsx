
import React, { MutableRefObject } from "react";

const scrollToRef = (ref: MutableRefObject<null | HTMLDivElement>) => {
    if (ref.current) {
        ref.current!.scrollIntoView({
            behavior: "smooth",
          });
        // On mobile Chrome smooth is not supported, so let's 
        // make sure we're scrolled anyway
        // ref.current!.scrollIntoView(true);
    }
};

const delay = (milliseconds: number) => {
    return new Promise(resolve => setTimeout(resolve, milliseconds));
}

const ReactHelpers = {
    scrollToRef, delay
}

export default ReactHelpers;