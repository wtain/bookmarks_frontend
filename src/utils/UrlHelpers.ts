
export function processUrls(text: string, visitor: (url: string) => string): string {
  const regex = /(https?:\/\/[^/]+(\/[\w-]*)+)/gm;

  return text.replaceAll(regex, (substring: string) => {
    return visitor(substring);
  });
}

export function processTextAndEnableLinks(text: string) {
  // todo: load target title, validate url works
  // todo: security concerns?
  return processUrls(text, (url: string) => {
    return `<A HREF='${url}'>${url}</A>`;
  })
    .replaceAll("\n", "<BR>");
}