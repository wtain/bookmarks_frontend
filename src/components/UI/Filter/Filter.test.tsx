import renderer from 'react-test-renderer';
import Filter from './Filter';
import MockTagsRepository from '../../../domain/repository/tags/MockTagsRepository';
import { BrowserRouter } from 'react-router-dom';
import { AppProvider } from "@shopify/polaris";
import translations from "@shopify/polaris/locales/en.json";
import { jestHacks } from '../../../utils/JestUtils';

describe('Filter tests', () => {
  beforeAll(() => {
    jestHacks();
  });

  it('Filter renders correctly', () => {
    const tagsRepository = new MockTagsRepository();

    const tree = renderer
      .create(
        <BrowserRouter>
          <AppProvider i18n={translations}>
            <Filter tagsRepository={tagsRepository} />
          </AppProvider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});