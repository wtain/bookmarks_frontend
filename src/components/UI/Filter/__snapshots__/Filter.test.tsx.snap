// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders correctly 1`] = `
<span
  className="tag"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  ❌
</span>
`;
